examples/ExampleCableSpan.cpp

```diff
@@ -0,0 +1,267 @@
+/*-----------------------------------------------------------------------------
+                Simbody(tm) Example: Cable Over Smooth Surfaces
+-------------------------------------------------------------------------------
+ Copyright (c) 2024 Authors.
+ Authors: Pepijn van den Bos
+ Contributors:
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may
+ not use this file except in compliance with the License. You may obtain a
+ copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ----------------------------------------------------------------------------*/
+
+/* This example is for experimenting with a CableSpan over obstacle surfaces.
+The cable endpoints are manually repositioned to control the next path
+solving problem. */
+
+#include "Simbody.h"
+#include "simbody/internal/CableSpan.h"
+#include <iostream>
+
+using namespace SimTK;
+
+// A helper class for printing interesting cable outputs.
+class ShowStuff : public PeriodicEventReporter {
+public:
+    ShowStuff(const CableSubsystem& cables, Real interval) :
+        PeriodicEventReporter(interval), m_cables(cables)
+    {}
+
+    static void showHeading(std::ostream& o)
+    {
+        printf(
+            "%10s %10s %10s %10s",
+            "Cable index",
+            "iterations",
+            "path error",
+            "length");
+    }
+
+    /** This is the implementation of the EventReporter virtual. **/
+    void handleEvent(const State& s) const override
+    {
+        for (CableSpanIndex cableIx(0); cableIx < m_cables.getNumCables();
+             ++cableIx) {
+            const CableSpan& cable = m_cables.getCable(cableIx);
+            printf(
+                "%d %d %g %g",
+                static_cast<int>(cableIx),
+                cable.getNumSolverIter(s),
+                cable.getMaxPathError(s),
+                cable.getLength(s));
+        }
+    }
+
+private:
+    const CableSubsystem& m_cables;
+};
+
+/* A helper class for drawing interesting things of a CableSpan. */
+class CableDecorator : public SimTK::DecorationGenerator {
+public:
+    CableDecorator(MultibodySystem& mbs, const CableSpan& cable) :
+        m_mbs(&mbs), m_cable(cable)
+    {
+        for (CableSpanObstacleIndex ix(0);
+             ix < m_cable.getNumSurfaceObstacles();
+             ++ix) {
+            m_obstacleDecorations.push_back(
+                m_cable.getObstacleContactGeometry(ix)
+                    .createDecorativeGeometry()
+                    .setResolution(3));
```

Comment:
@adamkewley does your comment https://github.com/simbody/simbody/pull/791#discussion_r1625670974 apply to this as well?